src/backend/joanie/payment/backends/lyra/__init__.py

```diff
@@ -331,7 +331,11 @@ def is_already_paid(self, order, installment):
         payload = {
             "orderId": str(order.id),
         }
-        response_json = self._call_api(url, payload)
+        try:
+            response_json = self._call_api(url, payload)
+        except exceptions.PaymentProviderAPIServerException:
+            order.set_installment_refused(installment["id"])
+            return False
```

Comment:
Hmmm we should instead catch `PaymentProviderAPIException` furthermore I'll do understand why you update the installment state to `refused`.

```suggestion
        try:
            response_json = self._call_api(url, payload)
        except exceptions.PaymentProviderAPIException:
            return False
```